components/sup/src/manager/service/mod.rs

```diff
@@ -77,26 +81,31 @@ impl ServiceSpec {
             topology: Topology::default(),
             update_strategy: UpdateStrategy::default(),
             binds: vec![],
-            dev_config_from: None,
+            config_from: None,
         }
     }
 }
 
 #[derive(Debug, Serialize)]
 pub struct Service {
-    pub needs_restart: bool,
-    pub package: Package,
     pub cfg_incarnation: u64,
-    pub service_group: ServiceGroup,
-    pub topology: Topology,
-    pub update_strategy: UpdateStrategy,
+    pub config: ServiceConfig,
+    pub config_root: PathBuf,
     pub current_service_files: HashMap<String, u64>,
+    pub depot_url: String,
+    pub health_check: HealthCheck,
     pub initialized: bool,
     pub last_restart_display: LastRestartDisplay,
-    pub supervisor: Supervisor,
-    pub depot_url: String,
+    pub needs_restart: bool,
+    pub package: Package,
+    pub service_group: ServiceGroup,
+    pub smoke_check: SmokeCheck,
     pub spec_ident: PackageIdent,
```

Comment:
Does `spec_ident` make sense to you? I needed a record of the "initial requirement" for the package to be booting into a service. In this way if you `hab start core/redis/1.2.3` we know how to check for updates more correctly. One interesting side effect of calling `hab start /path/to/core-redis-1.2.3-20170209010203-x86_64-linux.hart` is that `spec_ident` will be set to `core/redis/1.2.3/20170209010203` and would therefore *not* be updated. I think this is correct behavior albeit potentially surprising for a user on first blush.